src/cls/tabular/fbwriter_fbu.cc

```diff
@@ -0,0 +1,932 @@
+          db_schema_name,                   // db_schema_name TODO: parameterize
+          (uint32_t)nrows,                  // nrows
+          (uint32_t)ncols,                  // ncols
+          table_name,                       // table_name
+          delete_vector_fb,                 // delete_vector
+          Tables::Relation_FBU_Cols_FBU,    // relationData_type
+          cols.Union() ) ;                  // relationData
+
+        builder.Finish( root ) ;
+
+        // save each Col flatbuffer in one bufferlist and
+        // append the bufferlist to a larger bufferlist.
+        const char* dataptr = reinterpret_cast<char*>( builder.GetBufferPointer() ) ;
+        int datasz = builder.GetSize() ;
+        librados::bufferlist bl ;
+        bl.append( dataptr, datasz ) ;
+        ::encode( bl, bl_seq ) ;
+        buffer_size = buffer_size + datasz ;
+      }//if
+    } //for
+
+    std::cout << "buffer_size = " << buffer_size << std::endl ;
+
+    // --------------------------------------------- //
+    // build out FB_Meta
+    // --------------------------------------------- //
+    flatbuffers::FlatBufferBuilder *meta_builder = new flatbuffers::FlatBufferBuilder();
+    Tables::createFbMeta( meta_builder, 
+                          SFT_FLATBUF_UNION_COL,
+                          reinterpret_cast<unsigned char*>( bl_seq.c_str() ),
+                          bl_seq.length() ) ;
+
+    // add meta_builder's data into a bufferlist as char*
+    ceph::bufferlist meta_bl ;
+    char* meta_builder_ptr = reinterpret_cast<char*>( meta_builder->GetBufferPointer() ) ;
+    int meta_builder_size  = meta_builder->GetSize() ;
+    std::cout << "meta_builder_size = " << meta_builder_size << std::endl ;
+    meta_bl.append( meta_builder_ptr, meta_builder_size ) ;
+    delete meta_builder;
+    librados::bufferlist meta_wrapper_bl ;
+    ::encode( meta_bl, meta_wrapper_bl ) ;
+    size_t meta_wrapper_bl_sz = meta_wrapper_bl.length() ;
+    std::cout << "meta_wrapper_bl_sz = " << meta_wrapper_bl_sz << std::endl ;
+
+    // --------------------------------------------- //
+    // do the write
+    // --------------------------------------------- //
+    // write bufferlist
+    if( inputs.writeto == "ceph" )
+      //writeToCeph( bl_seq, buffer_size, inputs.targetformat, inputs.targetoid, inputs.targetpool ) ;
+      writeToCeph( meta_wrapper_bl, meta_builder_size, inputs.targetformat, inputs.targetoid, inputs.targetpool ) ;
+    else if( inputs.writeto == "disk" )
+      writeToDisk( bl_seq, buffer_size, inputs.targetformat, inputs.targetoid ) ;
+    else {
+      std::cout << ">>> unrecognized writeto '" << inputs.writeto << "'" << std::endl ;
+      exit( 1 ) ;
+    }
+
+  } //cols
+
+  // otherwise oops
+  else {
+    std::cout << ">> unrecognized inputs.write_type = '" << inputs.write_type << "'" << std::endl ;
+  }
+} // do_write
+
+std::string getSchemaString( uint64_t ncols, 
```

Comment:
You might be able to use our functions, but I'm not sure if that's what you want here.
https://github.com/uccross/skyhookdm-ceph/blob/skyhook-luminous/src/cls/tabular/cls_tabular_utils.cc#L904
https://github.com/uccross/skyhookdm-ceph/blob/skyhook-luminous/src/cls/tabular/cls_tabular_utils.cc#L937
https://github.com/uccross/skyhookdm-ceph/blob/skyhook-luminous/src/cls/tabular/cls_tabular_utils.cc#L897

Particularly the last one takes a schema string such as the lineitem test schema string (or sample schema from [file](https://github.com/uccross/skyhookdm-ceph/blob/skyhook-luminous/src/cls/tabular/SampleData/schema.txt)) and returns a schema vec with their col names and types etc.